pkg/test/framework/components/customsetup/setup.go

```diff
@@ -0,0 +1,277 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package customsetup
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/echo/common"
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/common/deployment"
+	depl "istio.io/istio/pkg/test/framework/components/echo/deployment"
+	"istio.io/istio/pkg/test/framework/components/echo/match"
+	"istio.io/istio/pkg/test/framework/components/util/cert"
+	"istio.io/istio/pkg/test/framework/resource"
+)
+
+type EchoDeployments struct {
```

Comment:
Why do we need a custom echo deployment? What is different about this and can we add it to the common one instead?

Reply:
There are some test specific instances with particular naming conventions, it's good if we add that here instead of adding it in main echo deployment? Will use it for other tests as well whenever it will be required.

Reply:
Can we instead change the tests to somehow just use the common echo deployment?

Reply:
okay, will work on this.